xcdat/temporal.py

```diff
@@ -728,17 +740,34 @@ def _set_data_var_attrs(self, data_var: str):
         self.data_var = data_var
         self.dim = get_dim_coords(dv, "T").name
 
+        first_time_coord = dv[self.dim].values[0]
+        is_decoded = isinstance(first_time_coord, cftime.datetime) or isinstance(
+            first_time_coord, np.datetime64
+        )
+        if not is_decoded:
+            raise ValueError(
+                f"The time coordinates type is {type(first_time_coord)}, not "
+                "'cftime.datetime' or 'np.datetime64'. Time coordinates must be "
+                "decoded as datetime before using TemporalAccessor methods. "
+            )
+
+        # Get the `cftime` date type based on the CF calendar attribute.
+        # The date type is used to get the correct cftime.datetime sub-class
+        # type for creating new grouped time coordinates for averaging.
         try:
             self.calendar = dv[self.dim].encoding["calendar"]
-            self.date_type = get_date_type(self.calendar)
         except KeyError:
-            raise KeyError(
-                f"The 'calendar' encoding attribute is not set on the '{data_var}' "
-                f"time coordinate variable ({self.dim}). This might indicate that the "
-                "time coordinates were not decoded, which is required for temporal "
-                "averaging operations. "
+            self.calendar = "standard"
+            logger.warning(
+                f"'{self.dim}' does not have a calendar encoding attribute set, "
+                "which is used to determine the `cftime.datetime` object type for the "
+                "output time coordinates. Defaulting to CF 'standard' calendar. "
+                "Otherwise, set the calendar type (e.g., "
+                "ds['time'].encoding['calendar'] = 'noleap') and try again."
             )
 
+        self.date_type = get_date_type(self.calendar)
```

Comment:
If no `calendar` attr is set, calendar is defaulted to `"standard"`. Calendar is required for determining the `cftime.datetime` object type, and checking if leap days should be dropped for high freq climatology/departures calculations.